engine/src/main/java/org/terasology/math/ChunkMath.java

```diff
@@ -0,0 +1,251 @@
+
+        int size = (maxX - minX + 1) * (maxY - minY + 1) * (maxZ - minZ + 1);
+
+        Vector3i[] result = new Vector3i[size];
+        int index = 0;
+        for (int x = minX; x <= maxX; x++) {
+            for (int y = minY; y <= maxY; y++) {
+                for (int z = minZ; z <= maxZ; z++) {
+                    result[index++] = new Vector3i(x, y, z);
+                }
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Returns the internal position of a block within a chunk.
+     *
+     * @param blockX The X-coordinate of the block in the world
+     * @return The X-coordinate of the block within the chunk
+     */
+    public static int calcBlockPosX(int blockX, int chunkPosFilterX) {
+        return blockX & chunkPosFilterX;
+    }
+
+    public static int calcBlockPosY(int blockY, int chunkPosFilterY) {
+        return blockY & chunkPosFilterY;
+    }
+
+    /**
+     * Returns the internal position of a block within a chunk.
+     *
+     * @param blockZ The Z-coordinate of the block in the world
+     * @return The Z-coordinate of the block within the chunk
+     */
+    public static int calcBlockPosZ(int blockZ, int chunkPosFilterZ) {
+        return blockZ & chunkPosFilterZ;
+    }
+
+    public static Vector3i calcBlockPos(Vector3i worldPos) {
+        return calcBlockPos(worldPos.x, worldPos.y, worldPos.z, ChunkConstants.INNER_CHUNK_POS_FILTER);
+    }
+
+    public static Vector3i calcBlockPos(int x, int y, int z) {
+        return calcBlockPos(x, y, z, ChunkConstants.INNER_CHUNK_POS_FILTER);
+    }
+
+    public static Vector3i calcBlockPos(int x, int y, int z, Vector3i chunkFilterSize) {
+        return new Vector3i(calcBlockPosX(x, chunkFilterSize.x), calcBlockPosY(y, chunkFilterSize.y), calcBlockPosZ(z, chunkFilterSize.z));
+    }
+
+    public static Region3i getChunkRegionAroundWorldPos(Vector3i pos, int extent) {
+        Vector3i minPos = new Vector3i(-extent, -extent, -extent);
+        minPos.add(pos);
+        Vector3i maxPos = new Vector3i(extent, extent, extent);
+        maxPos.add(pos);
+
+        Vector3i minChunk = calcChunkPos(minPos);
+        Vector3i maxChunk = calcChunkPos(maxPos);
+
+        return Region3i.createFromMinMax(minChunk, maxChunk);
+    }
+
+    // TODO: This doesn't belong in this class, move it.
+    public static Side getSecondaryPlacementDirection(Vector3f direction, Vector3f normal) {
+        Side surfaceDir = Side.inDirection(normal);
+        Vector3f attachDir = surfaceDir.reverse().getVector3i().toVector3f();
+        Vector3f rawDirection = new Vector3f(direction);
+        float dot = (float) rawDirection.dot(attachDir);
+        rawDirection.sub(new Vector3f(dot * attachDir.x, dot * attachDir.y, dot * attachDir.z));
+        return Side.inDirection(rawDirection.x, rawDirection.y, rawDirection.z).reverse();
+    }
+
+    /**
+     * Produces a region containing the region touching the side of the given region, both in and outside the region.
```

Comment:
Yeah, that is a confusing description. Probably also wrong.
I believe if you have a 10x10x10 region, the result is a 10x1x1 (or 1x10x1 or 1x1x10) region of the positions on one side of the region, just inside the region.